docs/recommendations/carousel/carousel.md

```diff
@@ -0,0 +1,72 @@
+# Carousel Props
+
+* [component](#component)
+* [loaderComponent](#loadercomponent)
+* [getComponentProps](#getcomponentprops)
+
+### component
+The props for your component comes from [getComponentProps](#getcomponentprops). This component renders each item in the carousel.
+
+#### Example
+```tsx
+function CustomComponent({ title, image }: any) {
+    return (
+        <div className="breinify-recommendation">
+            <div className="br-rec-content">
+                <div className="br-product-name">{title}</div>
+                <img className="br-product-image" src={image} />
+            </div>
+        </div>
+    );
+}
+
+export const CarouselTest = () => {
+    return <Carousel component={CustomComponent} />;
+};
+```
+### loaderComponent
+Customize the loader component
+
+#### Example
+```tsx
+function CustomLoader({ title, image }: any) {
+    return <div>Loading</div>;
+}
+
+export const CarouselTest = () => {
+    return <Carousel loaderComponent={CustomLoader} />;
+};
+```
+
+### getComponentProps
+A function to filter props passed down to the `component`. `getComponentProps` receives all the props and filters it based on user needs.
```

Comment:
I think a more accurate description for `getComponentProps` would be something like

```suggestion
A prop getter function to get the props that will be passed to `component`. `getComponentProps` is called with the one parameter: the recommendation result
```